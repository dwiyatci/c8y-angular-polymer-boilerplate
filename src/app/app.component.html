<!--<div>-->
<!--<label>Hello from Angular</label>-->
<!--<input [(ngModel)]="value">-->
<!--</div>-->
<!--<paper-input-->
<!--label="Hello from Polymer"-->
<!--ironControl-->
<!--[(ngModel)]="value"-->
<!--&gt;-->
<!--</paper-input>-->

<!--<div>-->
<!--<label>Non-form two-way bindings!</label>-->
<!--<input type="checkbox" [(value)]="checked" [(ngModel)]="checked">-->
<!--</div>-->
<!--<paper-checkbox-->
<!--[checked]="checked"-->
<!--(checked-changed)="checked = $event.detail.value"-->
<!--&gt;-->
<!--Checked-->
<!--</paper-checkbox>-->
<router-outlet></router-outlet>
