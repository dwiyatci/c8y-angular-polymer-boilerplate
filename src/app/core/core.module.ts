import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@c8y/ngx-components';

import { throwIfAlreadyLoaded } from './module-import-guard';
import { SelectivePreloadingStrategy } from './selective-preloading-strategy';

@NgModule({
  imports: [
    CommonModule
  ],
  providers: [
    SelectivePreloadingStrategy
  ],
  exports: [
    CommonModule
  ]
})
export class CoreModule {
  constructor(@Optional() @SkipSelf() parentModule: CoreModule) {
    throwIfAlreadyLoaded(parentModule, 'CoreModule');
  }
}
