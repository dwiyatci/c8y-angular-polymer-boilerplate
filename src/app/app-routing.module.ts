import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

import { AuthGuard } from './login/auth-guard.service';
import { SelectivePreloadingStrategy } from './core/selective-preloading-strategy';
import { AboutComponent } from './shared/about.component';
import { PageNotFoundComponent } from './shared/not-found.component';

const appRoutes: Routes = [
  {
    path: 'fiddle',
    loadChildren: './fiddle/fiddle.module#FiddleModule',
    canLoad: [AuthGuard]
  },
  {
    path: 'about',
    component: AboutComponent,
  },
  {
    path: '',
    redirectTo: '/fiddle',
    pathMatch: 'full'
  },
  {
    path: '**',
    component: PageNotFoundComponent
  }
];

@NgModule({
  imports: [
    RouterModule.forRoot(
      appRoutes,
      {
        // enableTracing: true, // <-- debugging purposes only
        useHash: true,
        preloadingStrategy: SelectivePreloadingStrategy
      }
    )
  ],
  exports: [
    RouterModule
  ]
})
export class AppRoutingModule {}
