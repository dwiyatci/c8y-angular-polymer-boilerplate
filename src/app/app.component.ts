import { Component } from '@angular/core';

// import '@polymer/paper-checkbox';
// import '@polymer/paper-input/paper-input.js';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  // value: string;
  // checked: boolean;
}
