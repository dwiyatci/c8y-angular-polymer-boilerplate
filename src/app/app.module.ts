import { BrowserModule } from '@angular/platform-browser';
import { CUSTOM_ELEMENTS_SCHEMA, NgModule } from '@angular/core';
import { PolymerModule } from '@codebakery/origami';
// import { OutletComponent } from '@c8y/ngx-components';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { CoreModule } from './core/core.module';
import { LoginModule } from './login/login.module';
import { SharedModule } from './shared/shared.module';

@NgModule({
  imports: [
    BrowserModule,
    SharedModule,
    CoreModule,
    PolymerModule.forRoot(),
    LoginModule,
    AppRoutingModule
  ],
  declarations: [
    AppComponent
  ],
  schemas: [CUSTOM_ELEMENTS_SCHEMA],
  bootstrap: [
    AppComponent,
    // OutletComponent
  ]
})
export class AppModule {}
