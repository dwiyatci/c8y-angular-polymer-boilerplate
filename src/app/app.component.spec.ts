import { DebugElement, NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { RouterLinkDirectiveStub } from '../testing/router-directive-link-stub';
import { configureTests } from '../testing/test-config.helper';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  it('can get RouterLinks from template', async(() => {
    setup().then(({ routerLinks }) => {
      expect(routerLinks.length)
        .toEqual(3);
      expect(routerLinks[0].linkParams)
        .toEqual('/');
      expect(routerLinks[1].linkParams)
        .toEqual('/fiddle');
      expect(routerLinks[2].linkParams)
        .toEqual('/about');
    });
  }));

  it('can click Fiddle link in template', async(() => {
    setup().then(({ fixture, linkDebugElts, routerLinks }) => {
      const fiddleLinkDebugElt = linkDebugElts[1]; // heroes link DebugElement
      const fiddleLink = routerLinks[1]; // heroes link directive

      expect(fiddleLink.navigatedTo)
        .toBeNull();

      fiddleLinkDebugElt.triggerEventHandler('click', null);

      expect(fiddleLink.navigatedTo)
        .toEqual('/fiddle');
    });
  }));

  test('renders markup to snapshot', () => {
    setup().then(({ fixture }) => {
      expect(fixture).toMatchSnapshot();
    });
  });

  function setup() {
    return configureTests(configure)
      .then((configuredTestBed) => {
        const fixture: ComponentFixture<AppComponent> = configuredTestBed.createComponent(AppComponent);
        const comp: AppComponent = fixture.componentInstance;

        fixture.detectChanges(); // trigger initial data binding

        // find DebugElements with an attached RouterLinkStubDirective
        const linkDebugElts: DebugElement[] = fixture.debugElement
          .queryAll(By.directive(RouterLinkDirectiveStub));

        // get attached link directive instances
        // using each DebugElement's injector
        const routerLinks: RouterLinkDirectiveStub[] = linkDebugElts.map(debugElt => debugElt.injector.get(RouterLinkDirectiveStub));

        return { fixture, comp, linkDebugElts, routerLinks };
      });
  }

  function configure(testBed) {
    testBed.configureTestingModule({
      declarations: [
        AppComponent,
        RouterLinkDirectiveStub
      ],
      schemas: [NO_ERRORS_SCHEMA]
    });
  }
});
