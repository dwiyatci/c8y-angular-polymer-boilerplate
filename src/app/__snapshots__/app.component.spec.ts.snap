// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent renders markup to snapshot 1`] = `
<app-root>
  <nav
    class="navbar navbar-expand-lg navbar-light"
    style="background-color: #e3f2fd;"
  >
    <a
      class="navbar-brand"
      ng-reflect-link-params="/"
      routerlink="/"
    >
      <img
        alt=""
        class="d-inline-block align-top"
        height="30"
        src="assets/img/icons8-violin-100.png"
        width="30"
      />
       Icycloutmu 
    </a>
    <div
      class="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-item nav-link"
          ng-reflect-link-params="/fiddle"
          routerlink="/fiddle"
        >
          Fiddle
        </a>
        <a
          class="nav-item nav-link"
          ng-reflect-link-params="/about"
          routerlink="/about"
        >
          About
        </a>
      </div>
    </div>
  </nav><router-outlet />
</app-root>
`;
